### PR TITLE
Add `.editorconfig` file

### PR DESCRIPTION
**Pull request by sobolevn**
CPython has one: https://github.com/python/cpython/blob/main/.editorconfig (I've used it as a source for inspiration)

It will also help us to have better editor support. Demo without this file:

https://user-images.githubusercontent.com/4660275/147847241-5a838089-76a1-4079-89b8-69bf058d7f0a.mov

Demo with this file:

https://user-images.githubusercontent.com/4660275/147847264-dc43cdc5-b954-472a-a668-ae64c162fa17.mov

### CODE REVIEW

**Review by Akuli — APPROVED** [review]
I like editorconfig files. I especially like how they can make pretty much any editor do things right, no matter how insane default behavior it has :)

I'll wait for another maintainer merge this, just in case someone wants to discuss.

**Review by JelleZijlstra — APPROVED** [review]
Seems useful!

**Review by srittau — APPROVED** [review]

**Pull request merged**
